sqlmesh/core/engine_adapter/bigquery.py

```diff
@@ -620,3 +633,51 @@ def should_retry(self, error: t.Type[Exception]) -> bool:
             )
             return True
         return False
+
+
+def select_partitions_expr(
+    schema: str,
+    table_name: str,
+    data_type: t.Union[str, exp.DataType],
+    granularity: t.Optional[str] = None,
+    agg_func: str = "MAX",
+    database: t.Optional[str] = None,
+) -> str:
+    """Generates a SQL expression that aggregates partition values for a table.
+
+    Args:
+        schema: The schema (BigQueyr dataset) of the table.
+        table_name: The name of the table.
+        data_type: The data type of the partition column.
+        granularity: The granularity of the partition. Supported values are: 'day', 'month', 'year' and 'hour'.
+        agg_func: The aggregation function to use.
+        database: The database (BigQuery project ID) of the table.
```

Comment:
call it catalog or projcet id

Reply:
I think we refer to this as `database` everywhere else. Don't want to introduce a new terminology, but doesn't hurt to clarify in comments.